hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetUtils.java

```diff
@@ -59,9 +64,12 @@ public static String normalize(String path) {
               + NetConstants.PATH_SEPARATOR_STR + ": " + path);
     }
 
+    if (len == 1 || path.charAt(len - 1) != NetConstants.PATH_SEPARATOR) {
```

Comment:
Can we simplified the logic without any matcher by using apache common StringUtils?

return len == 1 ? path : StringUtils.removeEnd(path, "/");

Reply:
@xiaoyuyao Thanks for review. But the result of  StringUtils.removeEnd("/a/b/c//", "/") is "/a/b/c/" instead of "/a/b/c". I'm not sure whether we should consider the path end with multiple "/".

Reply:
You are correct. The node to location mapping is provided by script or mapping file configured and parsed by implementation of DNSToSwitchMapping. The script should usually normalize the path already. Otherwise,  
the mapping can't  be used consistently for the topology class to determine node locations. 

E.g., multiple slash: "///" => ""
multiple slash in the middle "/a//b/c/" => "/a//b/c" which is not equal to "/a/b/c", etc.

Reply:
@xiaoyuyao Thanks for review. Updated.